providers/openfeature-provider-ofrep/src/openfeature/contrib/provider/ofrep/__init__.py

```diff
@@ -1,49 +1,180 @@
 
-    def resolve_string_details(  # type: ignore[empty-body]
+    def resolve_string_details(
         self,
         flag_key: str,
         default_value: str,
         evaluation_context: Optional[EvaluationContext] = None,
-    ) -> FlagResolutionDetails[str]: ...
+    ) -> FlagResolutionDetails[str]:
+        return self._resolve(flag_key, default_value, evaluation_context)
 
-    def resolve_integer_details(  # type: ignore[empty-body]
+    def resolve_integer_details(
         self,
         flag_key: str,
         default_value: int,
         evaluation_context: Optional[EvaluationContext] = None,
-    ) -> FlagResolutionDetails[int]: ...
+    ) -> FlagResolutionDetails[int]:
+        return self._resolve(flag_key, default_value, evaluation_context)
 
-    def resolve_float_details(  # type: ignore[empty-body]
+    def resolve_float_details(
         self,
         flag_key: str,
         default_value: float,
         evaluation_context: Optional[EvaluationContext] = None,
-    ) -> FlagResolutionDetails[float]: ...
+    ) -> FlagResolutionDetails[float]:
+        return self._resolve(flag_key, default_value, evaluation_context)
 
-    def resolve_object_details(  # type: ignore[empty-body]
+    def resolve_object_details(
         self,
         flag_key: str,
         default_value: Union[dict, list],
         evaluation_context: Optional[EvaluationContext] = None,
-    ) -> FlagResolutionDetails[Union[dict, list]]: ...
+    ) -> FlagResolutionDetails[Union[dict, list]]:
+        return self._resolve(flag_key, default_value, evaluation_context)
+
+    def _resolve(
+        self,
+        flag_key: str,
+        default_value: Union[bool, str, int, float, dict, list],
+        evaluation_context: Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[Any]:
+        now = datetime.now(timezone.utc)
+        if self.retry_after and now <= self.retry_after:
+            raise GeneralError(
+                f"OFREP evaluation paused due to TooManyRequests until {self.retry_after}"
+            )
+        elif self.retry_after:
+            self.retry_after = None
+
+        try:
+            response = self.session.post(
+                urljoin(self.base_url, f"/ofrep/v1/evaluate/flags/{flag_key}"),
+                json=_build_request_data(evaluation_context),
+                timeout=self.timeout,
+            )
+            response.raise_for_status()
+
+        except requests.RequestException as e:
+            self._handle_error(e)
+
+        try:
+            data = response.json()
+        except JSONDecodeError as e:
+            raise ParseError(str(e)) from e
+
+        return FlagResolutionDetails(
+            value=data["value"],
+            reason=Reason[data["reason"]],
+            variant=data["variant"],
+            flag_metadata=data["metadata"],
+        )
+
+    def _handle_error(self, exception: requests.RequestException) -> None:
```

Comment:
```suggestion
    def _handle_error(self, exception: requests.RequestException) -> NoReturn:
```
this fits better the purpose of the method, you need to import it from `typing`